extmod/vfs_lfsx.c

```diff
@@ -168,6 +173,7 @@ STATIC mp_obj_t MP_VFS_LFSx(ilistdir_it_iternext)(mp_obj_t self_in) {
         int ret = LFSx_API(dir_read)(&self->vfs->lfs, &self->dir, &info);
         if (ret == 0) {
             LFSx_API(dir_close)(&self->vfs->lfs, &self->dir);
+            self->vfs = NULL;
```

Comment:
This function will now segfault if you call `next(iter)` on the iterator after it already raised `StopIteration`.  It will need a check at the start for `self->vfs` being NULL, similar to the posix version.

Reply:
Ah, I was worried about exectly this but incorrectly thought a closed iterator wouldn't be able to be called on. I forgot about this syntax / usage.

Reply:
Thanks I also broadened the unit tests to cover this